app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulation.java

```diff
@@ -0,0 +1,169 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import java.util.Objects;
+import java.util.Set;
+
+import org.sagebionetworks.bridge.json.BridgeTypeName;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.Sets;
+
+/**
+ * A sub-population of the study participants who will receive a unique consent based on selection by a data group, 
+ * an application version, or both. 
+ */
+@DynamoDBTable(tableName = "Subpopulation")
+@BridgeTypeName("Subpopulation")
+public final class DynamoSubpopulation implements Subpopulation {
+
+    private String studyIdentifier;
+    private String guid;
+    private String name;
+    private String description;
+    private boolean required;
+    private boolean deleted;
+    private Integer minAppVersion;
+    private Integer maxAppVersion;
+    private Long version;
+    private Set<String> allOfGroups;
+    private Set<String> noneOfGroups;
+
+    public DynamoSubpopulation() {
+        this.allOfGroups = Sets.newHashSet();
+        this.noneOfGroups = Sets.newHashSet();
+    }
+    
+    @JsonIgnore
+    @Override
+    @DynamoDBHashKey
+    public String getStudyIdentifier() {
+        return studyIdentifier;
+    }
+    @Override
+    public void setStudyIdentifier(String studyIdentifier) {
+        this.studyIdentifier = studyIdentifier;
+    }
+    @Override
+    @DynamoDBRangeKey
+    public String getGuid() {
+        return guid;
+    }
```

Comment:
A few key design considerations:

* Is there any particular reason why study is the hash key and the guid is the range key? The guid uniquely identifies the subpop and hash lookups are much more efficient than range lookups. If the goal is to namespace guids within studies, we could always use a compound hash key. If we need to look up all subpops in a study, we could use a secondary index.

* Guids are researcher unfriendly. What if a researcher wants to look up a subpop by name? What if the researcher unintentionally creates two subpops with the same name? Would it be better to use name as the key instead of the guid (and restrict the name to URL-friendly JSON-friendly chars)?